### ISSUE TITLE
Setting Different Kubernetes API Endpoint for In-Cluster Deployment

### ISSUE THREAD

**Issue by exzenous** [kubernetes-sigs/headlamp#1460]
Hi,

I currently looking to deploy headlamp to a Google Kubernetes Engine (GKE) cluster as a In-Cluster deployment. While following the document to create token from a Service Account with appropriate role and role binding and use that token to authenticate through headlamp is working, I would like to allow users to authenticate with an OIDC IdP. (In my case, Keycloak)

Configuring GKE to allow users from an external IdP to authenticate to cluster (Almost the same as set `--oidc-xxx-yyy` flags for self-managed clusters)
can be done by following this document to add an external identity provider. 
[Use external identity providers to authenticate to GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/oidc)

Doing so will require users to change their Kubernetes API endpoint to `gke-oidc-envoy` service, which act as a proxy to the actual Kubernetes API.
![image](https://github.com/headlamp-k8s/headlamp/assets/33378035/e750f462-8704-463a-8f1d-cc9a5b343d7f)
(In my case, it is exposed to an Ingress)

Is there any configuration/environment variable that can be adjust to change the Kubernetes API endpoint? So that the token authenticated with Keycloak can be valid. 

Thank you.

**Comment by joaquimrocha** [comment]
@yolossn , can you take a look?

**Comment by yolossn** [comment]
Headlamp uses the `KUBERNETES_SERVICE_HOST`  and `KUBERNETES_SERVICE_PORT` env variables that are added by kubernetes by default for the In-cluster config, you can override those values by setting them in the Headlamp pod.

**Comment by yolossn** [comment]
@exzenous Were you able to solve this issue by following the above solution?

**Comment by exzenous** [comment]
@yolossn It seems to work! but right now I got certificate error instead.
![2023-10-31_07-03](https://github.com/headlamp-k8s/headlamp/assets/33378035/5b9909af-79a2-4493-bed3-5a73de9f0154)

It is strange that while directly accessing my own endpoint doesn't has this problem
![2023-10-31_07-07](https://github.com/headlamp-k8s/headlamp/assets/33378035/52dc5ddd-214a-4f67-bb67-6f784c65a013)

**Comment by yolossn** [comment]
Aah, I think this is because the kuberenetes go client is picking up the Token file and Root CA file from the default path. Do you have a Token and CA file?
https://github.com/kubernetes/client-go/blob/v0.28.3/rest/config.go#L511

**Comment by exzenous** [comment]
@yolossn Yes, I do have. I've tried to mount it to the path that kuberenetes go client is picking up but no dice. disable auto mounting service account causes an error
```
Failed to get in-cluster config open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
```
